### PR TITLE
[3.7] gh-101283: Improved fallback logic for subprocess with shell=True on Windows (GH-101286)

### PR DESCRIPTION
**Pull request by zooba**
* Issue: gh-101283

### CODE REVIEW

**Comment by zooba** [comment]
macOS crash is due to underlying OS update and not related to this PR.

**Comment by arhadthedev** [comment]
Undrafting since [the introduced bug](https://github.com/python/cpython/pull/101286#issuecomment-1423315974) will be fixed separately by backports of gh-101712. My apologizes for the confusion; I didn't sort out my incoming notifications timely.

File: Doc/library/subprocess.rst [hunk 1]
```diff
@@ -442,6 +450,17 @@ functions.
    :program:`ps`.  If ``shell=True``, on POSIX the *executable* argument
    specifies a replacement shell for the default :file:`/bin/sh`.
 
+   .. versionchanged:: 3.6
+      *executable* parameter accepts a :term:`path-like object` on POSIX.
```
Comment (arhadthedev) [inline Doc/library/subprocess.rst:454]: Was it added by mistake?

File: Doc/library/subprocess.rst [hunk 2]
```diff
@@ -111,6 +111,14 @@ compatibility with older versions, see the :ref:`call-function-trio` section.
       Added the *text* parameter, as a more understandable alias of *universal_newlines*.
       Added the *capture_output* parameter.
 
+   .. versionchanged:: 3.11.2
```
Comment (arhadthedev) [inline Doc/library/subprocess.rst]:
```suggestion
   .. versionchanged:: 3.7.17
```

File: Doc/library/subprocess.rst [hunk 3]
```diff
@@ -442,6 +450,17 @@ functions.
    :program:`ps`.  If ``shell=True``, on POSIX the *executable* argument
    specifies a replacement shell for the default :file:`/bin/sh`.
 
+   .. versionchanged:: 3.6
+      *executable* parameter accepts a :term:`path-like object` on POSIX.
+
+   .. versionchanged:: 3.11.2
```
Comment (arhadthedev) [inline Doc/library/subprocess.rst]:
```suggestion
   .. versionchanged:: 3.7.17
```

File: Doc/library/subprocess.rst [hunk 4]
```diff
@@ -1032,6 +1051,14 @@ calls these functions.
    .. versionchanged:: 3.3
       *timeout* was added.
 
+   .. versionchanged:: 3.11.2
```
Comment (arhadthedev) [inline Doc/library/subprocess.rst]:
```suggestion
   .. versionchanged:: 3.7.17
```

File: Doc/library/subprocess.rst [hunk 5]
```diff
@@ -1062,6 +1089,14 @@ calls these functions.
...
```
Comment (arhadthedev) [inline Doc/library/subprocess.rst]:
```suggestion
   .. versionchanged:: 3.7.17
```

File: Doc/library/subprocess.rst [hunk 6]
```diff
@@ -1116,6 +1151,14 @@ calls these functions.
    .. versionadded:: 3.7
       *text* was added as a more readable alias for *universal_newlines*.
 
+   .. versionchanged:: 3.11.2
```
Comment (arhadthedev) [inline Doc/library/subprocess.rst]:
```suggestion
   .. versionchanged:: 3.7.17
```

**Review by arhadthedev — COMMENTED** [review]
According to https://www.python.org/downloads/ the next published 3.7 release will be 3.7.17.

File: Doc/library/subprocess.rst (hunk 1 above)
Comment (ambv) [reply Doc/library/subprocess.rst:454]: It was probably added to resolve the conflict but it belongs in the 3.7 docs as well. For some reason this doc bit wasn't ported back to 3.6 and 3.7 when Serhiy landed GH-5914.

I would leave it in.

**Pull request merged**